sdk/src/Services/DynamoDBv2/Custom/DocumentModel/Search.cs

```diff
@@ -533,10 +560,12 @@ internal async Task<List<Document>> GetRemainingHelperAsync(CancellationToken ca
 
             while (!IsDone)
             {
+                var previousScannedCount = scannedCount;
```

Comment:
Once `GetNextSetHelper` is updated to always do `+=` for scanned count which makes it do the accumulation then I think the get remaining won't need any logic because it will all happen in `GetNextSetHelper`.

Reply:
My comment doesn't work since we decided `ScannedCount` does not accumulated for each `GetX` call.